pkg/controller/kieapp/defaults/defaults_test.go

```diff
@@ -5951,3 +5951,80 @@ func TestCRServerCPULimitAndRequestUsingMilicores(t *testing.T) {
 	assert.Equal(t, values.Requests.Cpu(), env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Resources.Requests.Cpu())
 	assert.Equal(t, values.Limits.Cpu(), env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Resources.Limits.Cpu())
 }
+
+func TestRhpamEnvironmentWithKafkaJBPM(t *testing.T) {
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{Name: "testJbpmEmitter"},
+		Spec: api.KieAppSpec{
+			Environment: api.RhpamProductionImmutable,
+			Objects: api.KieAppObjects{
+				Servers: []api.KieServerSet{
+					{Kafka: createKafkaExtObject(), KafkaJbpmEventEmitters: createKafkaJbpmObject()},
+				},
+			},
+		},
+	}
+
+	testEnvironmentWithKafkaJBPM(t, cr)
+}
+
+func testEnvironmentWithKafkaJBPM(t *testing.T, cr *api.KieApp) {
+	env, err := GetEnvironment(cr, test.MockService())
+	assert.NotNil(t, env)
+	assert.Nil(t, err, "Error getting environment")
+
+	for _, env := range env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Env {
+
+		checkJbpmKafkaEnvs(t, env)
+	}
+
+	assert.Equal(t, "true", getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_KAFKA_EXT_ENABLED"))
+	assert.Equal(t, "3", getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_KAFKA_JBPM_EVENT_EMITTER_ACKS"))
+	assert.Equal(t, "localhost:9092", getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_KAFKA_JBPM_EVENT_EMITTER_BOOTSTRAP_SERVERS"))
+	assert.Equal(t, "D12345678", getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_KAFKA_JBPM_EVENT_EMITTER_CLIENT_ID"))
+	assert.Equal(t, "2000", getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_KAFKA_JBPM_EVENT_EMITTER_MAX_BLOCK_MS"))
+	assert.Equal(t, "dd-MM-yyyy'T'HH:mm:ss.SSSZ", getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_KAFKA_JBPM_EVENT_EMITTER_DATE_FORMAT"))
+}
+
+func checkJbpmKafkaEnvs(t *testing.T, env corev1.EnvVar) {
+	switch e := env.Name; e {
+
+	case "KIE_SERVER_KAFKA_JBPM_EVENT_EMITTER_ACKS":
+		assert.Equal(t, env.Value, "3")
+
+	case "KIE_SERVER_KAFKA_JBPM_EVENT_EMITTER_BOOTSTRAP_SERVERS":
+		assert.Equal(t, env.Value, "localhost:9092")
+
+	case "KIE_SERVER_KAFKA_JBPM_EVENT_EMITTER_CLIENT_ID":
+		assert.Equal(t, env.Value, "D12345678")
+
+	case "KIE_SERVER_KAFKA_JBPM_EVENT_EMITTER_MAX_BLOCK_MS":
+		assert.Equal(t, env.Value, "2000")
+
+	case "KIE_SERVER_KAFKA_JBPM_EVENT_EMITTER_DATE_FORMAT":
+		assert.Equal(t, env.Value, "dd-MM-yyyy'T'HH:mm:ss.SSSZ")
+
+	case "KIE_SERVER_KAFKA_JBPM_EVENT_EMITTER_TASKS_TOPIC_NAME":
+		assert.Equal(t, env.Value, "my-tasks-topic")
+
+	case "KIE_SERVER_KAFKA_JBPM_EVENT_EMITTER_CASE_TOPIC_NAME":
+		assert.Equal(t, env.Value, "my-case-topic")
+
+	case "KIE_SERVER_KAFKA_JBPM_EVENT_EMITTER_PROCESS_TOPIC_NAME":
```

Comment:
is it correct? shouldn't the envs being set?

Reply:
Are set here https://github.com/kiegroup/kie-cloud-operator/pull/654/files#diff-20d788a31791c7601d275bf50c8e30cb790834f4c05ce64c4e3bdfe92a6aaebaR6025-R6027